### PR TITLE
RHINENG-7061: add "intersystems" field to schema

### PR DESCRIPTION
**Pull request by JoySnow**
[RHINENG-7061](https://issues.redhat.com/browse/RHINENG-7061) 

Puptoo system_profile PR: https://github.com/RedHatInsights/insights-puptoo/pull/316

### CODE REVIEW

**Review by kruai — APPROVED** [review]

**Comment by kruai** [comment]
@JoySnow This looks good, and I appreciate the link to the puptoo PR. Should I merge, or do you want to wait until the puptoo PR is approved as well?

**Review by kruai — CHANGES_REQUESTED** [review]
Oh, I just saw that there's a comment on [RHINENG-7061](https://issues.redhat.com/browse/RHINENG-7061) that says "iris" should be changed to "intersystems". Feel free to ping me for another review when that's been updated.

File: schemas/system_profile/v1.yaml [hunk 1]
```diff
@@ -602,6 +602,35 @@ $defs:
             maxLength: 12
             enum: ['Premium', 'Standard', 'Self-Support']
             example: 'Premium, Standard, Self-Support'
+      iris:
```
Comment (thearifismail) [inline schemas/system_profile/v1.yaml:605]: `iris` is one of 4 products in [Inersystems](https://issues.redhat.com/browse/RHINENG-7061?focusedId=23777789&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23777789).  Making `iris` a top-level element in system_profile does not look right.  Even `Intersystem` should fall under some super category in system_profile.  If there is no potential of having more siblings of `Intersystem`, then how about: 
```suggestion
    interSystems:
      category/product:
        description: A subcategory of interSystems
        type: object
        enum: [iris, healthshare, healthconnect, irishealth]
    iris:
      description: Object containing data specific to InterSystems workload
      type: object
      properties:
        iris_system:
          description: Indicates if InterSystems is installed on the system
          type: boolean
        iris_running_instances:
          type: array
          items:
            description: The info for each running InterSystems instances in the system
            type: object
            properties:
              instance_name:
                description: The name of the instance
                type: string
                example: 'IRIS3, PROD'
                maxLength: 255
              product:
                description: The product of the instance
                type: string
                example: 'IRIS'
                maxLength: 64
              version:
                description: The version of the instance
                type: string
                example:  '2023.1, 2023.2'
                maxLength: 7
                pattern: '^\d+(\.\d+)+$'
```

File: tests/utils/invalids.py [hunk 2]
```diff
@@ -135,6 +135,34 @@
     {"operating_system": {"name": "RHEL", "major": "9", "minor": "0"}},  # Invalid types
     {"katello_agent_running": "False"},
     {"satellite_managed": "True"},
+    {"iris": {  # Incorrect iris_system value
```
Comment (thearifismail) [inline tests/utils/invalids.py:138]: these should change if the suggestion above is accepted.

**Review by thearifismail — COMMENTED** [review]
Please see the comments

**Comment by JoySnow** [comment]
Thanks for the review, @kruai , @thearifismail .  

Got a new comment on [RHINENG-7061](https://issues.redhat.com/browse/RHINENG-7061) mentioned that "iris" should be changed to "intersystems". 

And I agree that would be a good approach. Confirming with the "initiator" in [RHINENG-7061](https://issues.redhat.com/browse/RHINENG-7061?focusedId=23814998&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-23814998 ) for the final  data format. 

We can hang up the review here and I will update the code accordingly later.

**Comment by JoySnow** [comment]
Updated the code accordingly after the field name changes confirmation in [RHINENG-7061](https://issues.redhat.com/browse/RHINENG-7061) .
(Amended the commit message to replace the "iris" word with "intersystems" in it.)

Hi @kruai , @thearifismail , could you kindly help review the changes?

**Review by thearifismail — APPROVED** [review]
LGTM

**Review by kruai — APPROVED** [review]

**Comment by kruai** [comment]
FYI @JoySnow I've modified the schema's format a bit (in #126). There should be no difference in the data structure or usage, just in the schema format.

**Pull request merged**